### PR TITLE
chore(installer): Remove the amazon-ecr-credential-helper dependency

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
- Updates installer E2E tests to use `installtesting.datad0g.com` instead of the QA registry
- Removes now unused jobs that push to the QA OCI registry
- Removes now unused `amazon-ecr-credential-helper` dependency

### Motivation
Removing the need for ECR auth in the installer to shrink the binary sizes

### CODE REVIEW

File: .gitlab/deploy_packages/oci.yml [hunk 1]
```diff
@@ -24,9 +24,10 @@ include:
     - datadog-package push registry.ddbuild.io/ci/remote-updates/${OCI_PRODUCT}:${VERSION} ${OMNIBUS_PACKAGE_DIR}/${OCI_PRODUCT}-${VERSION}.oci.tar
     # This is used for E2E tests. Doesn't cost more than an additional tag to the registry.
     - datadog-package push registry.ddbuild.io/ci/remote-updates/${OCI_PRODUCT}:pipeline-${CI_PIPELINE_ID} ${OMNIBUS_PACKAGE_DIR}/${OCI_PRODUCT}-${VERSION}.oci.tar
-    # Used for install scripts e2e tests
+    # Used for e2e tests
     - datadog-package replicate-s3 registry.ddbuild.io/ci/remote-updates/${OCI_PRODUCT}:pipeline-${CI_PIPELINE_ID} us-east-1 ${INSTALLER_TESTING_S3_BUCKET} ${S3_PACKAGE} ${VERSION}
     - datadog-package replicate-s3 registry.ddbuild.io/ci/remote-updates/${OCI_PRODUCT}:pipeline-${CI_PIPELINE_ID} us-east-1 ${INSTALLER_TESTING_S3_BUCKET} ${S3_PACKAGE} ${CI_COMMIT_SHA}
+    - datadog-package replicate-s3 registry.ddbuild.io/ci/remote-updates/${OCI_PRODUCT}:pipeline-${CI_PIPELINE_ID} us-east-1 ${INSTALLER_TESTING_S3_BUCKET} ${S3_PACKAGE} pipeline-${CI_PIPELINE_ID}
```
Comment (raphaelgavache) [inline .gitlab/deploy_packages/oci.yml:30]: oups, you can replace ${VERSION} on line 28 by pipelineID, it's my mistake, should have been pipelineID

Comment (BaptisteFoy) [reply .gitlab/deploy_packages/oci.yml:30]: I think it's fine to keep it, that way prod & staging registries are fully embedded in the testing one

**Review by raphaelgavache — APPROVED** [review]

**Comment by pducolin** [comment]
You could also keep images in the QA registry and install the ecr credential helper on the remote machine under test on EC2 using https://github.com/DataDog/test-infra-definitions/blob/main/scenarios/aws/ec2/vm.go#L113

**Review by pducolin — APPROVED** [review]

**Review by chouquette — APPROVED** [review]

**Pull request merged**